### PR TITLE
Add AppInsights heartbeat support to JobRunner

### PR DESCRIPTION
**Pull request by xavierdecoster**
This PR consumes ServerCommon v2.56.0 packages which depend on AppInsights v2.10 and adds heartbeat support to AppInsights initialization logic.

### CODE REVIEW

File: src/NuGet.Jobs.Common/JobRunner.cs [hunk 1]
```diff
@@ -234,6 +251,11 @@ private static string PrettyPrintTime(double milliSeconds)
                     _logger.LogInformation("Job run ended...");
                     stopWatch.Stop();
                     _logger.LogInformation("Job run took {RunDuration}", PrettyPrintTime(stopWatch.ElapsedMilliseconds));
+
+                    ApplicationInsights.HeartbeatManager?.SetHeartbeatProperty(
```
Comment (loic-sharma) [inline src/NuGet.Jobs.Common/JobRunner.cs]: Out of curiosity, how can we see a job's heart beat on App Insights' portal?

Comment (xavierdecoster) [reply src/NuGet.Jobs.Common/JobRunner.cs]: Np! :) 

At this point, all I know (and verified) is that the AI heartbeats end up in a separate metric in the AI portal. I believe we can easily query this and export to Geneva, but I haven't had a chance to verify this. The heartbeats should contain all info we need to identify the job and machine the heartbeat was sent from, but I also know this is extensible, so if anything is missing, we can add it.

File: src/StatusAggregator/StatusAggregator.csproj [hunk 2]
```diff
@@ -161,7 +161,7 @@
       <Version>2.55.0</Version>
     </PackageReference>
     <PackageReference Include="NuGet.Services.Status">
-      <Version>2.55.0</Version>
+      <Version>2.56.0</Version>
```
Comment (skofman1) [inline src/StatusAggregator/StatusAggregator.csproj]: why is this change needed?

File: src/NuGet.Services.Revalidate/NuGet.Services.Revalidate.csproj [hunk 3]
```diff
@@ -118,7 +118,7 @@
       <PrivateAssets>all</PrivateAssets>
     </PackageReference>
     <PackageReference Include="NuGet.Services.Status">
-      <Version>2.55.0</Version>
+      <Version>2.56.0</Version>
```
Comment (skofman1) [inline src/NuGet.Services.Revalidate/NuGet.Services.Revalidate.csproj]: why is this change needed?

File: src/StatusAggregator/StatusAggregator.csproj (hunk 2 above)
Comment (skofman1) [reply src/StatusAggregator/StatusAggregator.csproj]: if we are bumping all ServerCommon dependencies to 2.56, shouldn't we bump up NuGet.Services.Incidents, NuGet.Services.Status.Table?

---
In reply to: [328256095](https://github.com/NuGet/NuGet.Jobs/pull/808#discussion_r328256095) [](ancestors = 328256095)

File: src/NuGet.Jobs.Common/JobRunner.cs [hunk 4]
```diff
@@ -199,7 +210,13 @@ private static string PrettyPrintTime(double milliSeconds)
             var stopWatch = new Stopwatch();
             Stopwatch timeSinceInitialization = null;
 
-            int exitCode;
+            int exitCode = 0;
+
+            ApplicationInsights.HeartbeatManager?.AddHeartbeatProperty(
+                HeartbeatProperty_JobLoopExitCode,
```
Comment (skofman1) [inline src/NuGet.Jobs.Common/JobRunner.cs]: can u explain what this does?

File: src/StatusAggregator/StatusAggregator.csproj (hunk 2 above)
Comment (xavierdecoster) [reply src/StatusAggregator/StatusAggregator.csproj]: Technically, we only require to bump NuGet.Services.Logging to v2.56.0 to unlock Heartbeats and AI upgrade requirement of v2.10.

I figured it'd be good to align all NuGet.Services.* to v2.56.0? (though clearly I missed some...)

File: src/NuGet.Jobs.Common/JobRunner.cs (hunk 4 above)
Comment (xavierdecoster) [reply src/NuGet.Jobs.Common/JobRunner.cs]: In essence, this tells AppInsights that, even though a heartbeat is reported, the state of the application is unhealthy when the exitcode is different from zero.

Imagine the scenario in which the job loops continuously but every loop fails, the heartbeats don't know about any of this, and would report as healthy.

File: src/StatusAggregator/StatusAggregator.csproj (hunk 2 above)
Comment (skofman1) [reply src/StatusAggregator/StatusAggregator.csproj]: Got it! It will be great to have this as a comment in the code

File: src/NuGet.Jobs.Common/JobRunner.cs (hunk 4 above)
Comment (skofman1) [reply src/NuGet.Jobs.Common/JobRunner.cs]: Got it! It will be great to have this as a comment in the code

Comment (xavierdecoster) [reply src/NuGet.Jobs.Common/JobRunner.cs]: Done! https://github.com/NuGet/NuGet.Jobs/pull/808/commits/676664b29e672e2197633438cce8145e5da9ba19

File: src/StatusAggregator/StatusAggregator.csproj (hunk 2 above)
Comment (xavierdecoster) [reply src/StatusAggregator/StatusAggregator.csproj]: Upgraded other ServerCommon dependencies to align to v2.56.0 in https://github.com/NuGet/NuGet.Jobs/pull/808/commits/54393cc558d8b2f997015b7add7c4c4389f318ea

File: src/NuGet.Services.Revalidate/NuGet.Services.Revalidate.csproj (hunk 3 above)
Comment (xavierdecoster) [reply src/NuGet.Services.Revalidate/NuGet.Services.Revalidate.csproj]: See https://github.com/NuGet/NuGet.Jobs/pull/808#discussion_r328277398
